docs_src/websockets/tutorial003.py

```diff
@@ -42,24 +44,39 @@
 </html>
 """
 
+_T = TypeVar("_T")
+
+@attr.s(frozen=True, slots=True, auto_attribs=True)
+class _IdEq(Generic[_T]):
+    v: _T
+    def __eq__(self, other: object):
+        return isinstance(other, _IdEq) and self.v is other.v
+
+    def __hash__(self):
+        return hash(id(self.v))
+
 
 class ConnectionManager:
     def __init__(self):
-        self.active_connections: List[WebSocket] = []
+        self.active_connections: MutableSet[_IdEq(WebSocket)] = set()
```

Comment:
```suggestion
        self.active_connections: MutableSet[_IdEq[WebSocket]] = set()
```

Reply:
I've suggested this, but you probably want to create a variable for it. Because on the add you want to use that new variable to something like: `_newIdEq(websocket)`

Reply:
Might be better to drop the __eq__ from HTTPConnecttion

Reply:
@Kludex how did this pass the tests?